blogo-input/posts/votedelegation.md

```diff
@@ -0,0 +1,105 @@
+Observe that in this framework we can capture all delegation mechanisms. For instance, the predicate $f$ can just check that the identifier $id$ corresponds to the identifiers of the elections running in 2022 and 2023 or can check that $v$ belongs to a subset of options.
+
+
+## Snark-based anoynmous e-voting schemes
+Before describing our proposal to add delegation, we present a construction of a typical SNARK-based anonymous e-voting scheme. The following description is generic and does not describe a specific implementation.
+
+A typical anonymous e-voting scheme from SNARKs look as follows.
+
+*Setup Phase.* Let $n$ be the number of eligible voters. For each $i=1,\ldots,n$, we assume voter $i$ registers in the system with a public-key $ok_i$ for the signature scheme and retains privately the corresponding secret-key $sk_i$.
+All public-keys $(pk_1,\ldots,pk_n)$ of all eligible voters are published on a blockchain.
+
+The setup algorithm builds a Merkle tree $M$ (called the *census*) from $(pk_1,\ldots,pk_N)$ and publishes it on the blockchain. Let us call $R$ the root of $M$. The setup procedure outputs $R$ as public-key of the e-voting scheme. Note that the voter's public-keys are the leaves of $M$.
+
+We assume a public identifier $id$ is chosen for this election and published on the blockchain The string $id$ can be for instance the day in which the election is run along with other information about the election.
+
+*Cast Phase.* Each eligible voter $V$ who decides to cast a vote $v$ proceeds as follows.
+
+Let $v$ be the intended vote that $V$ wants to cast. The string $v$ can be both a valid voting option (e.g., the name of a candidate) or an arbitrary string representing an invalid vote. 
+
+Let $sk_V$ be the secret-key of $V$ corresponding to public-key $pk_V$, and let $h_V=H(sk_V,id).$
+
+Let $p$ be the Merkle-path in $M$ that proves that $R$ is the root of a tree that has $pk_V$ as leaf.
+Note that the length of $p$ is logarithmic in $M$.
+
+Let $C^{R,v,h_V,id}$ be the following Boolean circuit with one output gate.
+
+$C^{R,v,h_v,id}$ depends on the constants $R,v,h_V,id$, takes as input a pair $w=(p,sk_V)$ and outputs $1$ if and only if *all* the following conditions are verified:
+
+1. The string $p$ is a Merkle-path from $R$ to $pk_V$.
+2. $sk_V$ is a secret-key corresponding to the public-key $pk_V$. 
+3. $h_v=H(sk_V,id)$.
+
+Note that the values $R,v,h_V,id$, and thus $C^{R,v,h_v,id}$, will represent public information while $w$ is only known to $V$.
+The voter $V$ use the SNARK prover to compute a proof $\pi_V$ of the fact that $C^{R,v,h_v,id}$ is satisfied by witness $w$.
+
+$V$ publishes on the blockchain her ballot $B_V=(v,h_V,\pi_V)$.
+*Discarding invalid votes and duplicates.*
+When a voter sends a ballot $(v,h_V,\pi_V)$ to the PBB, we assume that if the proof $\pi_V$ is not accepted by the SNARK verifier, an error symbol $\bot$ is written next to the ballot to indicate that $v$ should not be considered for the tally.
```

Comment:
The acronym PBB should be defined on its first use

Reply:
I guess that this is solved by @vincenzoiovino last commit https://github.com/aragon/research-blog/pull/3/commits/a2ab24314c86b7f8a6a980aa3b57dd5fe339b14e